app/src/main/java/io/github/novacrypto/qrscanner/demo/DemoActivity.kt

```diff
@@ -44,11 +44,20 @@ class DemoActivity : AppCompatActivity() {
         launch.setOnClickListener {
             startActivityForResult(Intent(this, ScanQrActivity::class.java).apply {
                 putExtra(ScanQrActivity.OPTION_SHOW_BARCODE_BOX, BuildConfig.DEBUG)
+                putExtra(ScanQrActivity.AUTOACCEPT_RESULT, false)
+            }, REQUEST_SCAN)
+        }
+
+        // Auto-accept scanner
+        launch_autoaccept.setOnClickListener {
+            startActivityForResult(Intent(this, ScanQrActivity::class.java).apply {
+                putExtra(ScanQrActivity.OPTION_SHOW_BARCODE_BOX, BuildConfig.DEBUG)
+                putExtra(ScanQrActivity.AUTOACCEPT_RESULT, true)
```

Comment:
:+1:

Idea for future syntax:

```
startActivityForResult(Intent(this, ScanQrActivity::class.java)
    .withScannerOptions {
        if (BuildConfig.DEBUG) showBox()
        autoAccept()
    }, REQUEST_SCAN)
```

Then these ugly CAPITAL options and the room for error in the values passed to them disappears.

```
fun Intent.withScannerOptions(function: (IntentOptions)-> Unit) =
    function(IntentOptions(this))

class IntentOptions(private intent: Intent) {
    fun autoAccept(){
        intent.putExtra(ScanQrActivity.AUTOACCEPT_RESULT, true)
    }
    //etc
}
```

You get the idea? Would be separate issue and PR.

Reply:
I see, This is a good use of extension functions.